tern/utils/general.py

```diff
@@ -182,3 +183,175 @@ def parse_image_string(image_string):
                 'digest_type': tokens[1],
                 'digest': tokens[2]}
     return None
+
+
+def split_shell_script(script, skip_branch=False):
+    """Given a shell script, split it into statement of
+    1. variable
+    2. command
+    3. branch: if, case
+    Use dictionary to store infomations on a statement, it has following keys:
+    - variable(only for variable statement):
+        - name: variable name
+        - value: variable value
+    - variable(only for variable statement):
+        - name: command name(eg. apt-get)
+        - options: command options(eg. ["--no-install-recommends", "-y"])
+        - words: command words
+    - branch(only for variable statement):
+        - type: branch type(if or case)
+        - statements: branch variable and command statements list
+    - content: statement sentence(eg. 'apt-get update')
+
+    sentence is the text we are parsing.(string)
+    statement is the parsed result for sentence.(dictionary)
+    return a list of statements
+    """
+    # add ';' in case that the script is not ended with ';'
+    script += ';'
+    if skip_branch:
+        lexer = script
+    else:
+        # set posix to False to keep quotes.
+        lexer = shlex.split(script, posix=False)
+    sentence = []
+    statements = []
+    statement = {}
+    # ':;' stands for true
+    remove_token = [':;', '{', '}']
+    keywords = {'if': 'fi', 'case': 'esac'}
+    separators = ('&&', '||', ';')
+    # priority: keywords > remove_token > separator
+    # if come across a keyword(if or case), ignore remove_token and separator
+    # until meet the ending keyword(fi or esac)
+    # we will parse the keywords sentence in other functions
+    for tk in lexer:
+        # 1. check if in a branch
+        if sentence and sentence[0] in keywords.keys():
+            sentence.append(tk)
+            if tk.startswith(keywords[sentence[0]]):
+                if skip_branch:
```

Comment:
Here we will skip branches in branch sentences, which means we will ignore nested branches.